### PR TITLE
Fix: Adjusting plugin.json schema regex

### LINKED ISSUE #54445: Plugin Schema - Adjust version and grafanaDependency regex
Adjust plugin.schema.json regex patterns.

To fix:
- https://github.com/grafana/grafana/blob/main/docs/sources/developers/plugins/plugin.schema.json#L149 should also match minor versions >= 10 e.g. `>=8.4.10` (Possible regex to fix it `^(<=|>=|<|>|=|~|\^)?([0-9]+)(\.[0-9x\*]+)(\.[0-9x\*]+)?(\s(<=|>=|<|=>)?([0-9]+)(\.[0-9x\*]+)(\.[0-9x\*]+))?$`)
- https://github.com/grafana/grafana/blob/main/docs/sources/developers/plugins/plugin.schema.json#L306 should allow `%VERSION%`  (Possible regex to fix it: `^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)|(%VERSION%)$`)

### PR DESCRIPTION
**Pull request by tolzhabayev**
**What this PR does / why we need it**:
Fixes regex we use to validate `plugin.json`

- grafanaDepdenency
- - Adding support for minor version bigger than 9. E.g. `>=8.1.10`
- version
- - Some plugins use `%VERSION%` as value for `version` field. It did not validate properly before, now it does.

**Which issue(s) this PR fixes**:

Fixes #54445

**Special notes for your reviewer**:
- I am not sure how to test that the validation in `coremodel.cue` validates correctly, I did not find a test that I could extend.

### CODE REVIEW

File: pkg/coremodel/pluginmeta/coremodel.cue [hunk 1]
```diff
@@ -142,7 +142,7 @@ seqs: [
 
 					// Required Grafana version for this plugin. Validated using
 					// https://github.com/npm/node-semver.
-					grafanaDependency: =~"^(<=|>=|<|>|=|~|\\^)?([0-9]+)(\\.[0-9x\\*]+)(\\.[0-9x\\*])?(\\s(<=|>=|<|=>)?([0-9]+)(\\.[0-9x]+)(\\.[0-9x]))?$"
+					grafanaDependency: =~"^(<=|>=|<|>|=|~|\\^)?([0-9]+)(\\.[0-9x\\*]+)(\\.[0-9x\\*]+)?(\\s(<=|>=|<|=>)?([0-9]+)(\\.[0-9x]+)(\\.[0-9x]))?$"
```
Comment (wbrowne) [inline pkg/coremodel/pluginmeta/coremodel.cue:145]: Should this exactly match the `plugin.schema.json`? Seems it only adds the `+` for this one

Comment (tolzhabayev) [reply pkg/coremodel/pluginmeta/coremodel.cue:145]: Oof! good point, copy paste... I will fix the plugins.schema.json one it should only add the `+`

Comment (tolzhabayev) [reply pkg/coremodel/pluginmeta/coremodel.cue:145]: https://github.com/grafana/grafana/pull/54515/commits/bc4d2675c71672bc52f7cb4233549fcadefebc38

**Review by wbrowne — APPROVED** [review]

**Review by briangann — APPROVED** [review]
lgtm!

**Review by brendamuir — APPROVED** [review]

**Pull request merged**